src/transformers/trainer.py

```diff
@@ -3831,6 +3831,16 @@ def evaluation_loop(
             if is_torch_xla_available():
                 xm.mark_step()
 
+            if self.args.eval_use_gather_object and not is_accelerate_available("0.30.0"):
+                logger.warning(
+                    "You are using eval_use_gather_object = True with a version of `accelerate` < 0.30.0. This is not supported"
+                    " and we recommend you to update your version."
+                )
```

Comment:
Instead I would verify this with an exception in TrainingArguments post init - we want this to fail and fail as early as possible

Reply:
@SunMarc I also agree with this idea. What do you think about?

Reply:
@amyeroberts @SunMarc 
https://github.com/huggingface/transformers/pull/31514/commits/b47b0bd3300f2ef27e3b4ffed9b5ff9e21ba2c7e

Since I have moved the checking part to initial part of the trainer.py (which is `def create_accelerator_and_postprcoess`). I have some question.
There are some warning and caution function in this same level so I wanted to make sure should we make this warning or exception into the TrainingArguments or leave as-is. Could you clarify? (IMO, I agree with @amyeroberts)

Reply:
Let's go with what @amyeroberts suggested and raise an error